### PR TITLE
Add early stop to autoencoder

### PR DESCRIPTION
**Pull request by efajardo-nv**
PR from @gbatmaz originally targeting https://github.com/efajardo-nv/dfencoder/tree/morpheus-22.08.

### CODE REVIEW

File: dfencoder/autoencoder.py [hunk 1]
```diff
@@ -622,7 +622,7 @@ def _create_stat_dict(self, a):
         std = scaler.std
         return {'scaler': scaler, 'mean': mean, 'std': std}
 
-    def fit(self, df, epochs=1, val=None):
+    def fit(self, df, epochs=1, val=None, patience=5):
```
Comment (mdemoret-nv) [inline dfencoder/autoencoder.py]: Is `patience` something that will change for every call to `fit`? If not, we should be leaving the signature of `fit()` alone and change this to a contstructor arg.

File: dfencoder/autoencoder.py [hunk 2]
```diff
@@ -689,6 +692,25 @@ def fit(self, df, epochs=1, val=None):
...
+                    if self.verbose:
+                        print('The Current Net Loss:', current_net_loss)
+
+                    if current_net_loss > last_loss:
+                        count_es += 1
+                        if self.verbose:
+                            print('Early stop count:', count_es)
+
+                        if count_es >= patience:
+                            print('Early stopping: early stop count({}) >= patience({})'.format(count_es, patience))
```
Comment (mdemoret-nv) [inline dfencoder/autoencoder.py]: This needs to be surrounded by `if self.verbose`

**Review by mdemoret-nv — CHANGES_REQUESTED** [review]
Couple of small changes

File: dfencoder/autoencoder.py (hunk 1 above)
Comment (efajardo-nv) [reply dfencoder/autoencoder.py]: @mdemoret-nv discussed this with @gbatmaz. we'll move it to constructor to be more consistent with other training parameters like learning rate.

**Comment by efajardo-nv** [comment]
updates from feedback made. ready for re-review

**Review by mdemoret-nv — APPROVED** [review]

**Comment by mdemoret-nv** [comment]
@efajardo-nv Can you resolve the merge conflicts and then we can merge?

**Review by gbatmaz — APPROVED** [review]
LGTM

**Review by mdemoret-nv — APPROVED** [review]

**Review by gbatmaz — APPROVED** [review]
LGTM

**Pull request merged**